aws/resource_aws_glue_catalog_crawler.go

```diff
@@ -0,0 +1,411 @@
+	if len(targets) < 1 {
+		return []*glue.S3Target{}
+	}
+
+	perms := make([]*glue.S3Target, len(targets), len(targets))
+	for i, rawCfg := range targets {
+		cfg := rawCfg.(map[string]interface{})
+		perms[i] = expandS3Target(cfg)
+	}
+	return perms
+}
+
+func expandS3Target(cfg map[string]interface{}) *glue.S3Target {
+	target := &glue.S3Target{
+		Path: aws.String(cfg["path"].(string)),
+	}
+
+	if exclusions, ok := cfg["exclusions"]; ok {
+		target.Exclusions = expandStringList(exclusions.([]interface{}))
+	}
+	return target
+}
+
+func expandJdbcTargets(targets []interface{}) []*glue.JdbcTarget {
+	if len(targets) < 1 {
+		return []*glue.JdbcTarget{}
+	}
+
+	perms := make([]*glue.JdbcTarget, len(targets), len(targets))
+	for i, rawCfg := range targets {
+		cfg := rawCfg.(map[string]interface{})
+		perms[i] = expandJdbcTarget(cfg)
+	}
+	return perms
+}
+
+func expandJdbcTarget(cfg map[string]interface{}) *glue.JdbcTarget {
+	target := &glue.JdbcTarget{
+		Path:           aws.String(cfg["path"].(string)),
+		ConnectionName: aws.String(cfg["connection_name"].(string)),
+	}
+
+	if exclusions, ok := cfg["exclusions"]; ok {
+		target.Exclusions = expandStringList(exclusions.([]interface{}))
+	}
+	return target
+}
+
+func resourceAwsGlueCatalogCrawlerUpdate(d *schema.ResourceData, meta interface{}) error {
+	glueConn := meta.(*AWSClient).glueconn
+	name := d.Get("name").(string)
+
+	crawlerInput, err := createCrawlerInput(name, d)
+	if err != nil {
+		return err
+	}
+
+	crawlerUpdateInput := glue.UpdateCrawlerInput(*crawlerInput)
+	if _, err := glueConn.UpdateCrawler(&crawlerUpdateInput); err != nil {
+		return err
+	}
+
+	return resourceAwsGlueCatalogCrawlerRead(d, meta)
+}
+
+func resourceAwsGlueCatalogCrawlerRead(d *schema.ResourceData, meta interface{}) error {
+	glueConn := meta.(*AWSClient).glueconn
+	name := d.Get("name").(string)
+
+	input := &glue.GetCrawlerInput{
+		Name: aws.String(name),
+	}
+
+	crawlerOutput, err := glueConn.GetCrawler(input)
+	if err != nil {
+		if isAWSErr(err, glue.ErrCodeEntityNotFoundException, "") {
+			log.Printf("[WARN] Glue Crawler (%s) not found, removing from state", d.Id())
+			d.SetId("")
+		}
+
+		return fmt.Errorf("error reading Glue crawler: %s", err.Error())
+	}
+
+	d.Set("name", crawlerOutput.Crawler.Name)
+	d.Set("database_name", crawlerOutput.Crawler.DatabaseName)
+	d.Set("role", crawlerOutput.Crawler.Role)
+	d.Set("description", crawlerOutput.Crawler.Description)
+	d.Set("schedule", crawlerOutput.Crawler.Schedule)
+	d.Set("classifiers", flattenStringList(crawlerOutput.Crawler.Classifiers))
+	d.Set("table_prefix", crawlerOutput.Crawler.TablePrefix)
+
+	if crawlerOutput.Crawler.SchemaChangePolicy != nil {
+		schemaPolicy := map[string]string{
+			"delete_behavior": *crawlerOutput.Crawler.SchemaChangePolicy.DeleteBehavior,
+			"update_behavior": *crawlerOutput.Crawler.SchemaChangePolicy.UpdateBehavior,
+		}
+		d.Set("schema_change_policy", schemaPolicy)
```

Comment:
We should [perform error checking when working with aggregate attribute types](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types), e.g.

```go
if err := d.Set("schema_change_policy", schemaPolicy); err != nil {
  return fmt.Errorf("error setting schema_change_policy: %s")
}
```

I believe you'll find that its silently ignoring a type error currently because `schema_change_policy` is a `TypeList` and you're trying to set a `map`. You should be able to resolve this with something like:

```go
if err := d.Set("schema_change_policy", []map[string]interface{}{schemaPolicy}); err != nil {
  return fmt.Errorf("error setting schema_change_policy: %s")
}
```

(Although we generally prefer to leave typing logic like this in `flattenXXX()` functions)

Reply:
Done